hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/cache/topic/NotificationBasedTopicsCache.java

```diff
@@ -92,6 +94,11 @@ public void onTopicUnblacklisted(String qualifiedTopicName) {
     }
 
     @Override
+    public List<CachedTopic> getTopics() {
```

Comment:
so what you want to do is iterate through list, why not return immutable iterator instead fo copying the list around?